### PR TITLE
Feature/tictactoe

### PR DESCRIPTION
**Pull request by ninjanole5558**
add Tic Tac Toe. endGame() does not quite work with the API yet. Problem is with displaying the new window to select the play again option

### CODE REVIEW

File: src/arcade/games/tictactoe/tictactoe.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+        pg.Rect((self.pad,self.pad+self.sqDim),(self.square)),      #4 5 6
+        pg.Rect((self.pad+self.sqDim,self.pad+self.sqDim),(self.square)),
+        pg.Rect((self.pad+self.sqDim*2,self.pad+self.sqDim),(self.square)),
+
+        pg.Rect((self.pad,self.pad+self.sqDim*2),(self.square)),       #7 8 9
+        pg.Rect((self.pad+self.sqDim,self.pad+self.sqDim*2),(self.square)),
+        pg.Rect((self.pad+self.sqDim*2,self.pad+self.sqDim*2),(self.square))
+        ]
+
+    def onrender(self):
```
Comment (juliansweatt) [inline src/arcade/games/tictactoe/tictactoe.py:47]: I think this is why your end game window never shows up. ‘onrender()’ is called after PlethoraPy has rendered, and does not run again unless ‘onevent()’ returns True or ‘onrender()’ returns True. If you setup the endgame window within onrender and then return False, your window won’t ever update to show the endgame graphics.

Comment (dylnmc) [reply src/arcade/games/tictactoe/tictactoe.py:47]: You were right. I found the minor issues and corrected them

**Comment by dylnmc** [comment]
This looks ready to me. Since I am familiar with the API, I just quickly resolved the minor problems to make the end-game work with the API.

**Review by dylnmc — APPROVED** [review]
:+1: 

This looks ready to me. Since I am familiar with the API, I just quickly resolved the minor problems to make the end-game work with the API.

**Review by juliansweatt — APPROVED** [review]
Looks good to me!

**Pull request merged**